### PR TITLE
Remove unneeded plugins from IOs + some other janitor work

### PR DESCRIPTION
**Pull request by iemejia**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by iemejia** [comment]
This is some cleanup about issues I found while doing the last touches for HBaseIO. It is mostly repeated unneeded stuff, and some layout issues.
R: @davorbonaci  [I would normally put dhalperi but he should be tired of reviewing my PRs now :)].

File: sdks/java/javadoc/pom.xml [hunk 1]
```diff
@@ -84,16 +89,6 @@
 
     <dependency>
       <groupId>org.apache.beam</groupId>
-      <artifactId>beam-sdks-java-io-google-cloud-platform</artifactId>
```
Comment (davorbonaci) [inline sdks/java/javadoc/pom.xml]: I don't follow why this should be removed.

Comment (iemejia) [inline sdks/java/javadoc/pom.xml]: This and beam-sdks-java-core were repeated (they appear twice). Look at the final file, both are still there.

**Comment by iemejia** [comment]
It is explained now, I hope you are ok with it now. Thanks for the review.

**Pull request closed without merging**